plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java

```diff
@@ -1088,8 +1101,27 @@ public boolean isPageComplete() {
 
 	public IServer saveServer(IProgressMonitor monitor) throws CoreException {
 		model.updateServer();
+		setOpenshiftMavenProfileActive(model.getDeployProject());
 		return model.saveServer(monitor);
 	}
+	
+	@SuppressWarnings("restriction")
+    private void setOpenshiftMavenProfileActive(IProject newDeployProject)  throws CoreException  {
+	    IFile pom = newDeployProject.getFile(IMavenConstants.POM_FILE_NAME);
+	    if (newDeployProject.hasNature(IMavenConstants.NATURE_ID) && pom != null) {                   
+            final IProfileManager profileManager = MavenProfilesCoreActivator.getDefault().getProfileManager();
+            IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().create(pom, true, null);
+            List<ProfileData> profiles = profileManager.getProfileDatas(facade, null);
+            List<String> activeProfiles = profiles.stream()
+                    .filter(p -> ProfileState.Active.equals(p.getActivationState()))
+                    .map(p -> p.getId())
+                    .collect(Collectors.toList());
+            if (profiles.stream().anyMatch(p -> OPENSHIFT_MAVEN_PROFILE.equals(p.getId())) && !activeProfiles.contains(OPENSHIFT_MAVEN_PROFILE)) {
+                activeProfiles.add(OPENSHIFT_MAVEN_PROFILE);
+                profileManager.updateActiveProfiles(facade, activeProfiles, false, true, new NullProgressMonitor());
+            }
```

Comment:
l1111-1122 could be refactored to a set of 3 methods which then would be reusable - if then moved to a common bundle :
* ProfileData getActiveProfiles(IProject project)
* boolean isActiveProfile(String profileId, IProject project) - optional, could be included in setActiveProfile
* void setActiveProfile(String profileId)

Looks like a benefit if done so, agree?